File: src/libqthreadstorage.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://doc.qt.io/qt-6/qthreadstoragedata.html
pub const qthreadstoragedata = struct {
    /// New constructs a new QThreadStorageData object.
    ///
    /// ``` param1: QtC.QThreadStorageData ```
    pub fn New(param1: ?*anyopaque) QtC.QThreadStorageData {
        return qtc.QThreadStorageData_new(@ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qthreadstoragedata.html#get)
    ///
    /// ``` self: QtC.QThreadStorageData ```
    pub fn Get(self: ?*anyopaque) ?*anyopaque {
        return qtc.QThreadStorageData_Get(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qthreadstoragedata.html#set)
    ///
    /// ``` self: QtC.QThreadStorageData, p: ?*anyopaque ```
    pub fn Set(self: ?*anyopaque, p: ?*anyopaque) ?*anyopaque {
        return qtc.QThreadStorageData_Set(@ptrCast(self), p);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qthreadstoragedata.html#finish)
    ///
    /// ``` param1: ?*anyopaque ```
    pub fn Finish(param1: ?*anyopaque) void {
        qtc.QThreadStorageData_Finish(param1);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qthreadstoragedata.html#dtor.QThreadStorageData)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QThreadStorageData ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QThreadStorageData_Delete(@ptrCast(self));
    }
};

File: src/libqthreadstorage.cpp
#include <QThreadStorageData>
#include <qthreadstorage.h>
#include "libqthreadstorage.h"
#include "libqthreadstorage.hxx"

QThreadStorageData* QThreadStorageData_new(const QThreadStorageData* param1) {
    return new QThreadStorageData(*param1);
}

void** QThreadStorageData_Get(const QThreadStorageData* self) {
    return self->get();
}

void** QThreadStorageData_Set(QThreadStorageData* self, void* p) {
    return self->set(p);
}

void QThreadStorageData_Finish(void** param1) {
    QThreadStorageData::finish(param1);
}

int QThreadStorageData_Id(const QThreadStorageData* self) {
    return self->id;
}

void QThreadStorageData_SetId(QThreadStorageData* self, int id) {
    self->id = static_cast<int>(id);
}

void QThreadStorageData_Delete(QThreadStorageData* self) {
    delete self;
}
